.github/workflows/release.yml

```diff
@@ -0,0 +1,556 @@
+name: "Release Builds"
+on:
+  push:
+    branches:
+      - v[0-9].[0-9]**
+    tags:
+      - v[0-9].[0-9]**
+      - v[0-9].[0-9]**-**
+
+env:
+  MESON_VERSION: '0.56.2'
+  EM_VERSION: '2.0.29'
+  EM_CACHE_FOLDER: 'emsdk'
+  TAISEI_NOPRELOAD: 0
+  TAISEI_PRELOAD_REQUIRED: 1
+
+jobs:
+  linux-release-build-x64:
+    name: "Linux (x64/Source)"
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Tools
+      run: >
+        sudo apt update || true # debian cache is not atomic and can sometimes fail
+
+        sudo apt install -y -qq
+        build-essential
+        libsdl2-dev
+        libx11-dev
+        libwayland-dev
+        python3-docutils
+        gnupg
+
+        pip install
+        meson==${{ env.MESON_VERSION }}
+        ninja
+        zstandard
+        python-gnupg
+
+    # NOTE: confusingly, 'fetch-depth: 0' == "get all history", not "no history"
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+        fetch-depth: 1
+
+    - name: Import GPG Key
+      uses: taisei-project/ghaction-import-gpg@v3
+      with:
+        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+    - name: Set Build Env
+      id: vars
+      run: |
+        git fetch --force --tags # see: https://github.com/actions/checkout/issues/290
+        echo ::set-output name=tag::${GITHUB_REF#refs/tags/v}
+
+    - name: Configure
+      run: >
+        meson setup build/linux
+        --native-file misc/ci/linux-x86_64-build-release.ini
+        --prefix=$(pwd)/build-release
+
+    - name: Build
+      run: |
+        meson compile -C build/linux --verbose
+        meson install -C build/linux
+
+    - name: Run Test
+      run: $(pwd)/build-release/bin/taisei -R $(pwd)/misc/ci/tests/test-replay.tsr
+      env:
+        TAISEI_NOPRELOAD: ${{ env.TAISEI_NOPRELOAD }}
+        TAISEI_PRELOAD_REQUIRED: ${{ env.TAISEI_PRELOAD_REQUIRED }}
+
+    - name: Package (Binary)
+      run: ninja txz -C build/linux --verbose
+
+    - name: Upload Artifact (Binary)
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: linux-x86_64-${{ steps.vars.outputs.tag }}-txz
+        path: build/linux/Taisei-${{ steps.vars.outputs.tag }}-linux-x86_64.tar.xz*
+        if-no-files-found: error
+
+    - name: Package (Source)
+      run: >
+        meson rewrite kwargs set project / version v${{ steps.vars.outputs.tag }}
```

Comment:
`meson dist` goes off of the `project.version` set in `meson.build`, which then outputs files labeled `v1.4-dev` (currently) instead of whatever tag is being used in `git`. There might be some way around it, but this `rewrite` I have in there is intended as a workaround which then uses the tag from git, as expected.

Reply:
But then you'll still have the tarball's copy of `meson.build` using the outdated version number?

Really, the version number should be incremented in an actual commit, and that commit is what should be tagged.

Reply:
> But then you'll still have the tarball's copy of `meson.build` using the outdated version number?
> 
> Really, the version number should be incremented in an actual commit, and that commit is what should be tagged.

For an actual release I 100% agree with you. But this wouldn't work for a quick test run, which I presume this workaround is for. It is really dirty though; I'm quite unhappy with it. I didn't outright object to it because I don't have a better idea right now.